Tribler/Core/Modules/restapi/channels_endpoint.py

```diff
@@ -164,105 +94,65 @@ def render_GET(self, request):
-    @db_session
-    def render_POST(self, request):
-        personal_root = self.channel_id == 0 and self.channel_pk == self.session.lm.mds.my_key.pub().key_to_bin()[10:]
-        # TODO: better error handling
-        target_collection = self.session.lm.mds.CollectionNode.get(
-            public_key=database_blob(self.channel_pk), id_=self.channel_id
-        )
-        try:
-            request_parsed = json.twisted_loads(request.content.read())
-        except ValueError:
-            request.setResponseCode(http.BAD_REQUEST)
-            return json.twisted_dumps({"error": "Bad JSON"})
-
-        if not target_collection and not personal_root:
-            request.setResponseCode(http.NOT_FOUND)
-            return json.twisted_dumps({"error": "Target channel not found"})
-        results_list = []
-        for entry in request_parsed:
-            public_key, id_ = database_blob(unhexlify(entry["public_key"])), entry["id"]
-            source = self.session.lm.mds.ChannelNode.get(public_key=public_key, id_=id_)
-            if not source:
-                request.setResponseCode(http.BAD_REQUEST)
-                return json.twisted_dumps({"error": "Source entry not found"})
-            # We must upgrage Collections to Channels when moving them to root channel, and, vice-versa,
-            # downgrade Channels to Collections when moving them into existing channels
-            if isinstance(source, self.session.lm.mds.CollectionNode):
-                src_dict = source.to_dict()
-                if self.channel_id == 0:
-                    rslt = self.session.lm.mds.ChannelMetadata.create_channel(title=source.title)
+    async def copy_channel(self, request):
+        with db_session:
+            channel_pk, channel_id = self.get_channel_from_request(request)
+            personal_root = channel_id == 0 and channel_pk == self.session.lm.mds.my_key.pub().key_to_bin()[10:]
+            # TODO: better error handling
+            target_collection = self.session.lm.mds.CollectionNode.get(
+                public_key=database_blob(channel_pk), id_=channel_id
+            )
+            try:
+                request_parsed = await request.json()
+            except (ContentTypeError, ValueError):
+                return RESTResponse({"error": "Bad JSON"}, status=HTTP_BAD_REQUEST)
+
+            if not target_collection and not personal_root:
+                return RESTResponse({"error": "Target channel not found"}, status=HTTP_NOT_FOUND)
+            results_list = []
+            for entry in request_parsed:
+                public_key, id_ = database_blob(unhexlify(entry["public_key"])), entry["id"]
+                source = self.session.lm.mds.ChannelNode.get(public_key=public_key, id_=id_)
+                if not source:
+                    return RESTResponse({"error": "Source entry not found"}, status=HTTP_BAD_REQUEST)
+                # We must upgrage Collections to Channels when moving them to root channel, and, vice-versa,
```

Comment:
I know this is not your change, but could you please make this say "upgrade"?

Reply:
Will do.